python/python/lance/torch/data.py

```diff
@@ -56,6 +53,13 @@ def _to_tensor(
             )
             tensor = torch.from_numpy(np_tensor)
             del np_tensor
+        elif (isinstance(arr.type, pa.FixedShapeTensorType)) and (
+            pa.types.is_floating(arr.type.value_type)
```

Comment:
Can we use `arr.storage.values` to avoid copy?

```
>>> import numpy as np
>>> a = np.array(range(0, 9), dtype=np.float32).reshape(1, 3, 3)
>>> a
array([[[0., 1., 2.],
        [3., 4., 5.],
        [6., 7., 8.]]], dtype=float32)
>>> b = pa.FixedShapeTensorArray.from_numpy_ndarray(a)
>>> b
<pyarrow.lib.FixedShapeTensorArray object at 0x1031514e0>
[
  [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8
  ]
]
>>> b.storage
<pyarrow.lib.FixedSizeListArray object at 0x103168760>
[
  [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8
  ]
]
>>> b.storage.value
b.storage.value_counts()         b.storage.value_lengths()        b.storage.value_parent_indices() b.storage.values
>>> b.storage.values
<pyarrow.lib.FloatArray object at 0x103168880>
[
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8
]
```

Also it seems we missed a test case to cover fixed shaped tensor?

We can probably just do the following

```
elif isinstance(arr.type, pa.FixedShapedTensorType):
  return _to_tensor(arr.storage)
```

Reply:
I switched to using arr.storage (using a similar but slightly different simplification) and added a unit test.